api/utils/keys/yubikey_common.go

```diff
@@ -14,13 +14,40 @@ limitations under the License.
 package keys
 
 import (
+	"context"
+
 	"github.com/gravitational/trace"
 )
 
+// GetYubiKeyPrivateKey attempt to retrieve a YubiKey private key matching the given hardware key policy
+// from the given slot. If slot is unspecified, the default slot for the given key policy will be used.
+// If the slot is empty, a new private key matching the given policy will be generated in the slot.
+//   - hardware_key: 9a
+//   - hardware_key_touch: 9c
+func GetYubiKeyPrivateKey(ctx context.Context, policy PrivateKeyPolicy, slot string) (*PrivateKey, error) {
+	priv, err := getOrGenerateYubiKeyPrivateKey(ctx, policy, slot)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to get a YubiKey private key")
+	}
+	return priv, nil
+}
+
+// TODO(Joerger): Deprecated in favor of GetYubiKeyPrivateKey.
+// Delete once all references in /e are removed
 func GetOrGenerateYubiKeyPrivateKey(touchRequired bool) (*PrivateKey, error) {
-	priv, err := getOrGenerateYubiKeyPrivateKey(touchRequired)
+	policy := PrivateKeyPolicyHardwareKey
+	if touchRequired {
+		policy = PrivateKeyPolicyHardwareKeyTouch
+	}
+
+	priv, err := getOrGenerateYubiKeyPrivateKey(context.TODO(), policy, "")
```

Comment:
nit: Pass the context from the caller?

Reply:
`GetYubiKeyPrivateKey` takes ctx from the caller and is replacing this function. I can't change this function signature without breaking `/e`.